packages/state-transition/src/cache/epochCache.ts

```diff
@@ -426,8 +444,10 @@ export class EpochCache {
     return new EpochCache({
       config: this.config,
       // Common append-only structures shared with all states, no need to clone
-      pubkey2index: this.pubkey2index,
-      index2pubkey: this.index2pubkey,
+      finalizedPubkey2index: this.finalizedPubkey2index,
+      finalizedIndex2pubkey: this.finalizedIndex2pubkey,
+      // Fork-aware cache needs to be cloned. But by the virtue of it being persistent, we don't need to do anything here
```

Comment:
This statement sounds incorrect. Please add a unit test in the package to prove it:
- create state A, add pubkey 1
- clone state A to B, assert pubkey 1 is in A and B
- add pubkey 2 to B, assert pubkey 2 is in B but not in A

Reply:
> This statement sounds incorrect. Please add a unit test in the package to prove it:
> 
> * create state A, add pubkey 1
> * clone state A to B, assert pubkey 1 is in A and B
> * add pubkey 2 to B, assert pubkey 2 is in B but not in A

Updated

Reply:
Thanks! Tests look good. Then update the comment to something like below, since the UX of immutable is a bit different than the rest of persistent data structures in this repo
```
No need to clone this reference. On each mutation the `unfinalizedPubkey2index` reference is replaced, @see `addPubkey`
```